schematic/store/synapse.py

```diff
@@ -557,25 +558,36 @@ def getFilesInStorageDataset(
         file_list = []
 
         # iterate over all results
-        for dirpath, dirname, filenames in walked_path:
+        for dirpath, _, path_filenames in walked_path:
             # iterate over all files in a folder
-            for filename in filenames:
-                if (not "manifest" in filename[0] and not fileNames) or (
-                    fileNames and filename[0] in fileNames
+            for path_filename in path_filenames:
+                if ("manifest" not in path_filename[0] and not fileNames) or (
+                    fileNames and path_filename[0] in fileNames
                 ):
-                    # don't add manifest to list of files unless it is specified in the list of specified fileNames; return all found files
+                    # don't add manifest to list of files unless it is specified in the
+                    # list of specified fileNames; return all found files
                     # except the manifest if no fileNames have been specified
                     # TODO: refactor for clarity/maintainability
 
                     if fullpath:
                         # append directory path to filename
-                        filename = (
-                            project_name + "/" + dirpath[0] + "/" + filename[0],
-                            filename[1],
-                        )
+                        if dirpath[0].startswith(f"{project_name}/"):
+                            path_filename = (
+                                dirpath[0] + "/" + path_filename[0],
+                                path_filename[1],
+                            )
+                        else:
```

Comment:
I read unit test `test_getFilesInStorageDataset` again. I think that test only covers the "else" logic over here. Since the "if" logic gets introduced, could you also help modify `test_getFilesInStorageDataset`? Thank you!

Reply:
to be more specific, the `mock_return` in `test_getFilesInStorageDataset` mock the return of `walk`. But now it 
only cover cases where the project names are not included. Could you add cases where the project names are included? so that the "if" statement will get triggered/covered in the test.

Reply:
Added https://github.com/Sage-Bionetworks/schematic/pull/1474/commits/6642bf7e0048d115466cd40995ebaa4398004c96